RunningOrder/Shared/Components/InlineEditableLink.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  InlineEditableLink.swift
+//  RunningOrder
+//
+//  Created by Loic B on 26/02/2021.
+//  Copyright © 2021 Worldline. All rights reserved.
+//
+
+import SwiftUI
+
+struct InlineEditableLink: View {
+    @State private var hovered = false
+    @State private var isEditing: Bool
+    @State private var isDoneButtonEnabled = true
+    @ObservedObject var logic: Logic
+
+    /// - Parameters:
+    ///   - value: A binding to a Link Entity which referers to the Link url and label
+    init(value: Binding<Link>) {
+        self._isEditing = State(initialValue: value.wrappedValue.label.isEmpty)
+        self.logic = Logic(value: value)
+    }
+
+    var body: some View {
+        VStack(alignment: .leading) {
+            HStack {
+                if isEditing {
```

Comment:
I think it could be re-worked :
```swift
if let url = logic.value.formattedURL, ! isEditing {
   SwiftUI.Link(logic.value.label, destination: url)
   Spacer()
   if hovered {
      Button("Edit")  { self.isEditing = true }
          .buttonStyle(PlainButtonStyle())
          .foregroundColor(.accentColor)
   }
} else {
   VStack {
       BorderedTextField(placeholder: "Label", value: logic.$value.label)
       BorderedTextField(placeholder: "Url", value: logic.$value.url)
   }
   Spacer()
   Button("Done", action: logic.doneAction)
      .buttonStyle(PlainButtonStyle())
      .foregroundColor(.accentColor)
      .disabled(logic.isFieldEmpty)
}
```

Reply:
not now maybe, but later